test/cpp/interop/xds_interop_client.cc

```diff
@@ -345,6 +347,18 @@ class TestClient {
     std::unique_ptr<ClientAsyncResponseReader<SimpleResponse>>
         simple_response_reader;
   };
+  static bool RpcStatusCheckSuccess(AsyncClientCall* call) {
+    // Determine RPC success based on expected status.
+    if (absl::GetFlag(FLAGS_expect_status) == "OK" && call->status.ok()) {
```

Comment:
Please use `status_code_from_string()` here.

https://github.com/grpc/grpc/blob/64f04e4d2fad514c56bfd3a9145f6403c3697281/src/core/lib/channel/status_util.h#L31

Then you can just do something like this:

```
grpc_status_code code;
GPR_ASSERT(grpc_status_code_from_string(
    absl::GetFlag(FLAGS_expect_status).c_str(), &code));
return code == call->status.error_code();
```